emukit/core/constraints.py

```diff
@@ -0,0 +1,73 @@
+import logging
+from typing import Callable, Optional
+
+import numpy as np
+
+_log = logging.getLogger(__name__)
+
+
+class IConstraint:
+    def evaluate(self, x: np.ndarray) -> np.ndarray:
+        """
+        :param x: Input locations to evaluate constraint at
+        :return: Numpy array of shape (n_input,) where an element will be 1 if the corresponding input satisfies the
+                 constraint and zero if the constraint is violated
+        """
+        raise NotImplementedError
+
+
+class LinearInequalityConstraint(IConstraint):
+    """
+    Constraint of the form lower_bound < Ax < upper_bound where the matrix A is called "constraint_matrix"
+    """
+    def __init__(self, constraint_matrix: np.ndarray, lower_bound: np.ndarray, upper_bound: np.ndarray):
```

Comment:
I think this could use some defaults. Say identity matrix, -infinity, +infinity. As well as validation, that A!=0 and at least one bound is set.

Reply:
I'll add defaults for the bounds and some validation, I'm not keen on identity for constraint_matrix as you should be using the parameter bounds in that case rather than a constraint.